core/templates/dev/head/domain/collection/CollectionObjectFactory.js

```diff
@@ -203,6 +223,85 @@ oppia.factory('CollectionObjectFactory', [
       return skillList;
     };
 
+    // Gets a new ID for a skill. This should be of the same form as in the
+    // backend.
+    Collection.prototype.getNewSkillId = function() {
+      var newId = 's' + String(this._skillIdCount);
+      this._skillIdCount++;
+      return newId;
+    };
+
+    // Adds a new frontend collection skill domain object to this collection.
+    // This will return true if the skill was successfully added, or false if
+    // the given collection skill has an ID already used by another skill, the
+    // ID is invalid, or another skill already has the same name.
+    // Changes to the provided object will be reflected in this collection.
+    Collection.prototype.addCollectionSkill = function(collectionSkillObject) {
+      if (!this._skills.hasOwnProperty(collectionSkillObject.getId())) {
+        if (Number(collectionSkillObject.getId().slice(1)) >=
+            this._skillIdCount) {
+          return false;
+        }
+        for (var skill in this._skills) {
+          if (skill.name == collectionSkillObject.getName()) {
+            return false;
+          }
+        }
+        this._skills[collectionSkillObject.getId()] = collectionSkillObject;
+        return true;
+      }
+      return false;
+    };
+
+    // Attempts to remove a collection skill from this collection given the
+    // specified skill ID. Returns whether the collection skill was
+    // removed, which depends on whether any collection skills reference the
+    // given skill ID.
+    Collection.prototype.deleteCollectionSkill = function(skillId) {
+      if (this._skills.hasOwnProperty(skillId)) {
+        delete this._skills[skillId];
+        return true;
+      }
+      return false;
+    };
+
+    // Deletes all collection skills within this collection.
+    Collection.prototype.clearCollectionSkills = function() {
+      this._skills = {};
+      this.setSkillIdCount(0);
+    };
+
+    // Returns a dict mapping ids to collection skill objects for this
+    // collection.
+    Collection.prototype.getCollectionSkills = function() {
+      return this._skills;
+    };
+
+    // Returns if collection contains skill with given ID.
+    Collection.prototype.containsCollectionSkill = function(skillId) {
+      return this._skills.hasOwnProperty(skillId);
+    };
+
+    // Gets the ID of the skill with a given name. Returns null if skill name
+    // is not found.
+    Collection.prototype.getSkillIdFromName = function(skillName) {
+      for (var skillId in this._skills) {
+        if (this._skills[skillId].getName() == skillName) {
+          return skillId;
+        }
+      }
+      return null;
+    };
+
+    // Gets the skill object of from the collection by ID. Returns null if the
```

Comment:
Does this return null if the ID is invalid or if the collection does not contain a skill with the specified ID?